Write initial POSCARs as vasp5

Ash-Pera opened the pull request:
Made initial POSCAR write check if vasp5 and write appropriately. Also minor readability improvement in relax.py.

mdavezac commented on src/pylada/vasp/keywords.py:
```diff
@@ -874,7 +874,11 @@ def output_map(self, **kwargs):
         from ..crystal import write, read, specieset
         from . import files
-
+        try:
+            from pylada import is_vasp_4
+        except:
+            is_vasp_4 = False
+        
```
I think it would be better to simply add `is_vasp_4 = False` in config/vasp.py and then simply do

```suggestion
        from pylada import is_vasp_4
```

Looking back, that's what we should have done in the first place!

ftherrien commented:
@mdavezac any idea why the test fails with build (3.6) on Linux ? I ran the tests with python. 3.8 on Fedora and nothing went wrong

mdavezac commented:
Not sure yet, I have to investigate

On Sat, 23 May 2020, 15:22 Félix Therrien, <notifications@github.com> wrote:

> @mdavezac <https://github.com/mdavezac> any idea why the test fails with
> build (3.6) on Linux ? I ran the tests with python. 3.8 on Fedora and
> nothing went wrong
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pylada/pylada-light/pull/57#issuecomment-633063433>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAU6LCP4IUC3LFXXS4LN36TRS7L2RANCNFSM4NIB64EA>
> .
>

Ash-Pera commented:
I've added the `is_vasp_4` configuration variable, and changed the places in the code that used the try/except syntax. I do wonder, should `write.py` be updated to use this syntax? It currently uses 
```
if vasp5 is None:
        import pylada
        vasp5 = not getattr(pylada, 'is_vasp_4', True)
```
Is there a specific reason for that?

mdavezac commented:
No reason that I can remember. We could indeed get rid of the getattr and replace it with `vasp5 = not pylada.is_vasp_4`

Ash-Pera commented:
I'm a little bit confused by the automated test suite—it succeeds the MacOS tests but fails every single one of the Linux tests due to an import error. Not really sure how to interpret that, did I mess up something so much as to break the whole thing?

mdavezac commented:
I'm trying to figure it out, but I'm guessing it's not you ;)

I'd like to get them passing before we merge, out of principle. But if it
takes too long, we'll merge before that.

On Sun, 24 May 2020, 18:56 Ash Pera, <notifications@github.com> wrote:

> I'm a little bit confused by the automated test suite—it succeeds the
> MacOS tests but fails every single one of the Linux tests due to an import
> error. Not really sure how to interpret that, did I mess up something so
> much as to break the whole thing?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pylada/pylada-light/pull/57#issuecomment-633267334>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAU6LCNFC2KQAARSP4SPWWDRTFNTFANCNFSM4NIB64EA>
> .
>

mdavezac commented:
@ftherrien, any chance you can check that vasp stuff still works with this, and then merge? I don't have a license so I can not do it myself.
